src/arch_ppc.py

```diff
@@ -173,6 +177,71 @@ def match(self, matcher: AsmMatcher) -> Optional[Replacement]:
         return None
 
 
+class DoubleToIntIrPattern(IrPattern):
+    replacement = "cvt.i.d.ficitve $i, $f"
+    parts = [
+        "fctiwz $t, $f",
+        "stfd $t, N($r1)",
+        "lwz $i, (N+4)($r1)",
+    ]
+
+
+class CheckConstantMixin:
+    def check(self, m: IrMatch) -> bool:
+        # TODO: Also validate that `K($k)` is the expected constant in rodata
+        return m.symbolic_registers["k"] in m.arch.constant_regs
+
+
+class SintToDoubleIrPattern(IrPattern, CheckConstantMixin):
+    replacement = "cvt.d.i.fictive $f, $i, K($k)"
```

Comment:
This is a constraint on the pattern `parts`: because there's an instruction that reads from `K($k)` in the pattern, `K($k)` must be listed as one of the `inputs` to the replacement. 

When doing pattern replacement, it's possible to elide `outputs` by moving them into `clobbers`, this is what's done to the `$a`, `$b`, `N($r1)`, etc. terms. 
There isn't an equivalent concept for `inputs`. Allowing you to elide `inputs` would affect the soundness of the replacement, *and* require additional code to support.